pkg/scheduler/worker.go

```diff
@@ -64,14 +69,16 @@ func (wf *WorkersFactory) ForComponent(component string) (ReconciliationWorker,
 	if reconcilerCfg == nil {
 		return nil, fmt.Errorf("No reconciler found for component %s", component)
 	}
-	return NewWorker(reconcilerCfg, wf.mothershipHost, wf.mothershipPort, wf.inventory, wf.operationsReg, wf.debug)
+	return NewWorker(reconcilerCfg, wf.mothershipHost, wf.mothershipPort, wf.crdComponents, wf.preComponents, wf.inventory, wf.operationsReg, wf.debug)
```

Comment:
The parameters list become quite long, please maybe wrap those with some struct and just pass one parameter.